code/network/multiteamselect.cpp

```diff
@@ -1673,69 +1670,35 @@ void multi_ts_init_objnums()
 	}		
 }
 
-bool multi_ts_validate_ship(char *shipname, char *wingname) 
-{
-	int wing_number, wing_idx; 
-
-	// check name isn't too short to be valid
-	if (strlen(shipname) < (strlen (wingname) + 2) ) {		
-		return false;
-	}
-
-	wing_idx = wing_lookup(wingname); 
-	Assert (wing_idx >= 0 && wing_idx < MAX_WINGS); 
-	wing_number = atoi(shipname+strlen(wingname)); 
-
-	if (wing_number > 0 && wing_number <= Wings[wing_idx].wave_count) {
-		return true; 
-	}
-
-	return false;
-}
-
 // get the proper team and slot index for the given ship name
-void multi_ts_get_team_and_slot(char *ship_name,int *team_index,int *slot_index, bool mantis2757switch)
+void multi_ts_get_team_and_slot(char* ship_name, int* team_index, int* slot_index)
 {
-	int idx; 
 
 	// set the return values to default values
 	*team_index = -1;
 	*slot_index = -1;
 
+	// set up the ship registry pointer
+	const ship_registry_entry* ship_regp = ship_registry_get(ship_name);
+
+	// For a player usable ship, there has to be a parse object, a team and a position within a wing.
+	Assert(ship_regp != nullptr && ship_regp->p_objp != nullptr); 
+	if (ship_regp == nullptr || ship_regp->p_objp == nullptr) {
+		return;
+	}
+
+	*team_index = ship_regp->p_objp->team;
+
 	// if we're in team vs. team mode
 	if(Netgame.type_flags & NG_TYPE_TEAM){
-		Assert(MAX_TVT_WINGS == MULTI_TS_MAX_TVT_TEAMS);
-		for (idx = 0; idx < MAX_TVT_WINGS; idx++) {
-			// get team (wing)
-			if ( !strnicmp(ship_name, TVT_wing_names[idx], strlen(TVT_wing_names[idx])) && multi_ts_validate_ship(ship_name, TVT_wing_names[idx]) ) {				
-				*team_index = idx;
-				*slot_index = (ship_name[strlen(ship_name)-1] - '1');
-
-				// just Assert(), if this is wrong then we're pretty much screwed either way
-				Assert( (*slot_index >= 0) && (*slot_index < MAX_WSS_SLOTS) );
-			}
-		}
-	} 
-	// if we're _not_ in team vs. team mode
+		// get the slot within the wing, since there's only one wing each in team vs. team.
+		*slot_index = ship_regp->p_objp->pos_in_wing;
+	} // if we're _not_ in team vs. team mode
```

Comment:
I'd prefer the comment to be aligned with the `else` block, as it was before, but this is a nitpick.